### DISCUSSION TITLE
(Quasar + Vite + SPA) - Component level imports not split into separate chunks

### DISCUSSION THREAD

**Discussion by dtomlinson91** [quasarframework/quasar#14524] [General - Components / Directives / etc] [answered]
- quasar v2.8.3
- @quasar/app-vite v1.1.1
- vite v2.9.15

In one of my components I'm including mermaidjs:

```
<script setup lang="ts">
import mermaid from 'mermaid';

const props = defineProps<{
  productionChain: string;
  mermaidDefinition: string;
}>();

mermaid.render(
    props.productionChain,
    props.mermaidDefinition,
    (svgCode: string) => {
        // removed for brevity - sets ref
    }
  );
```

When building this gets bundled with the component js:

<img width="1071" alt="Screenshot 2022-09-30 at 09 34 16" src="https://user-images.githubusercontent.com/53234158/193249158-14e927c9-ea14-4b99-8e61-68259291701f.png">

Is this default behaviour for vite/rollup? I have tried:

```
extendViteConf(viteConf) {
    viteConf.build.rollupOptions = {
      output: {
        manualChunks(id) {
          if (id.includes('node_modules')) {
            return 'vendor';
          }
        },
      },
    };
  },
```

in `quasar.config.js` but this doesn't split the import into its own chunk.

If I move the import into a boot file (`boot/mermaid.ts`):

```
import { onMounted } from 'vue';
import mermaid from 'mermaid';

function initialiseMermaid() {
  onMounted(() => {
    mermaid.initialize({
      startOnLoad: false,
      logLevel: 'fatal',
      securityLevel: 'loose',
      theme: 'neutral',
      flowchart: { htmlLabels: true }
    });
  });
}

export { mermaid, initialiseMermaid };
```

it get's bundled into its own chunk:

<img width="1058" alt="Screenshot 2022-09-30 at 12 27 51" src="https://user-images.githubusercontent.com/53234158/193260237-f2ef43bb-84f8-4012-8a44-81998ef00a1f.png">

Is this intentional? Why doesn't `build.rollupOptions.output.manualChunks` have an effect on moving mermaid into its own chunk when imported in a component?

**Comment by rstoenescu** [comment, marked as answer]
This is intentional Vite behavior. And quite desired I might add. The end user gets to download exactly the code that it needs and nothing more.
However, if you want to create a vendor chunk, I'd point you to the Vite docs.
